pyccel/codegen/printing/ccode.py

```diff
@@ -853,11 +931,34 @@ def _print_AugAssign(self, expr):
         return "{0} {1}= {2};".format(lhs_code, op, rhs_code)
 
     def _print_Assign(self, expr):
+        lhs = self._print(expr.lhs)
+        rhs = expr.rhs
+        if isinstance(rhs, (NumpyArray)):
+            dummy_array_name = "dummy_%s" % (rhs.__hash__()%991)
+            dtype = self.find_in_dtype_registry(format(rhs.dtype), rhs.precision)
+            arg = rhs.arg
+            if rhs.rank > 1:
+                import functools
+                import operator
+                arg = functools.reduce(operator.concat, arg)
+            dummy_array = "%s %s[] = {%s};" % (dtype, dummy_array_name, ', '.join(self._print(i) for i in arg))
+            dtype = self.find_in_dtype_registry(format(rhs.dtype), rhs.precision, array=True)
+            cpy_data = "memcpy({0}.nd_{2}, {1}, {0}.buffer_size);".format(lhs, dummy_array_name, dtype)
```

Comment:
I don't know if this is possible but can we do:
```
a = np.array([1,2,3])
b = np.array(a)
```
? If it works in numpy then does it work here?